### PR TITLE
Jd update

### PR DESCRIPTION
**Pull request by eringh**
Quite some of the handling around deflation was previously done in the JD-code. But as of the deflation rewrite #167 that is no longer needed (the new code also seems to be more efficient). Hence that overhead is removed from JD.   (Also picks up an update from #111, when in changing in the JD-code)

### CODE REVIEW

**Comment by eringh** [comment]
Should I move the Schur-complement LinSolver for deflated NEPs to `LinSolvers`? As discussed in #167.

**Comment by jarlebring** [comment]
Wow. It really reduces the code a lot. Nice!

File: src/method_jd.jl [hunk 1]
```diff
@@ -257,18 +250,13 @@ function jd_effenberger(::Type{T},
     end
 
     conveig += 1
-    Λ[1,1] = λ
-    X[:,1] = u
-    deflated_nep = DeflatedGenericNEP(nep, Λ[1:conveig,1:conveig], X[:,1:conveig])
+    deflated_nep = deflate_eigpair(nep, λ, u)
```
Comment (jarlebring) [inline src/method_jd.jl]: Would it be possible to pass the deflation `mode` parameter as an optional parameter to the JD-call?

**Comment by jarlebring** [comment]
Yes. I think the Schur-complement linsolver can be moved. I changed my mind again: I think it belongs in `nep_deflation.jl` not LinSolvers. Maybe it should be the `default_linsolver` for deflated  `DeflatedNEP`?

**Comment by eringh** [comment]
It turns out we have some code overlap also in the linear systems. The function `compute_U` has in part the same functionality as some of the code in `compute_Mder` for deflated NEPs. I think we should align and try to work on code-reuse when moving the linear solver.

**Comment by jarlebring** [comment]
Yes. It sounds sensible to separate the U-part in the `compute_Mder(::DeflatedGenericNEP...)` (where it is referred to as `Q`) into a separate function and then call that function also in  `compute_U`.

**Comment by eringh** [comment]
After new discussion: We do not implement a generic Schur complement solver for saddle point matrices. The feature could turn out a bit complicated and is currently only needed in JD, where it is already implemented.

**Comment by eringh** [comment]
I feel ready to merge when the tests are done.
@jarlebring: What do you think?

**Comment by jarlebring** [comment]
It looks good. Ready for merge.

**Pull request merged**